src/test/java/hw7/jdi/JdiSiteTests.java

```diff
@@ -0,0 +1,65 @@
+package hw7.jdi;
+
+import com.epam.jdi.light.driver.WebDriverUtils;
+import com.epam.jdi.light.elements.init.PageFactory;
+import hw7.dataproviders.TestDataProvider;
+import hw7.jdi.entities.JdiUser;
+import hw7.jdi.entities.TestData;
+import org.testng.annotations.AfterSuite;
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Test;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Properties;
+
+import static java.lang.String.format;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+public class JdiSiteTests {
+
+    public static final String METALS_COLORS = "METALS & COLORS";
+    protected Properties userProperties;
+
+    @BeforeSuite(alwaysRun = true)
+    public void beforeSuite() {
+        PageFactory.initSite(JdiSite.class);
+        userProperties = new Properties();
+        try {
+            userProperties.load(new FileInputStream(
+                    "src/main/resources/jdiuser.properties"));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @AfterSuite
+    public void afterSuite() {
+        WebDriverUtils.killAllSeleniumDrivers();
+    }
+
+    @Test(dataProviderClass = TestDataProvider.class,
+            dataProvider = "Data Sets Provider")
+    public void metalsAndColorsPageFormTest(TestData testData) {
+        JdiUser jdiUser = new JdiUser(userProperties.getProperty("username"),
```

Comment:
This is not the best idea to generate user class from properties exactly in the test, chose an appropriate place for this.

Reply:
Solved: I removed these lines from test and replaced the test user instance in the `JdiUser`-class as a constant (just as in lessons). To get data from properties, I created `UserPropertiesSingleton`.